sql2gee/collection.py

```diff
@@ -37,10 +37,12 @@ def _where(self):
         It gets *where* conditions and converts them in the proper filters.
         self.asset.filter(ee.Filter([ee.Filter.eq('scenario','historical'), ee.Filter.date('1996-01-01','1997-01-01')])).filterBounds(geometry)
         """
-        if 'where' in self._parsed and self._parsed['where']:
-            if self.geometry:
+        if 'where' in self._parsed and len(self._parsed['where']) > 0:
+            if self.geometry and 'filter' in self._filters:
                 self._asset = self._asset.filter(self._filters['filter']).filterBounds(self.geometry)
-            else:
+            elif self.geometry:
+                self._asset = self._asset.filterBounds(self.geometry)
+            elif 'filter' in self._filters:
```

Comment:
what happens when `if not self.geometry and 'filter' not in self._filters` ?

Reply:
Nothing will be added which should be correct